be/src/serde/protobuf_serde.cpp

```diff
@@ -251,8 +277,31 @@ StatusOr<vectorized::Chunk> ProtobufChunkDeserializer::deserialize(std::string_v
             return Status::Corruption(fmt::format("mismatched row count: {} vs {}", col->size(), rows));
         }
     }
+
+    // deserialize extra data
+    vectorized::ChunkExtraDataPtr chunk_extra_data;
+    if (!_meta.extra_data_metas.empty()) {
+        std::vector<vectorized::ColumnPtr> extra_columns;
+        extra_columns.resize(_meta.extra_data_metas.size());
+        for (size_t i = 0, sz = _meta.extra_data_metas.size(); i < sz; ++i) {
+            auto extra_meta = _meta.extra_data_metas[i];
+            extra_columns[i] =
+                    ColumnHelper::create_column(extra_meta.type, extra_meta.is_null, extra_meta.is_const, rows);
+        }
+        for (auto& column : extra_columns) {
+            cur = ColumnArraySerde::deserialize(cur, column.get());
+        }
+        for (auto& col : extra_columns) {
+            if (col->size() != rows) {
+                return Status::Corruption(fmt::format("mismatched row count: {} vs {}", col->size(), rows));
+            }
+        }
+        chunk_extra_data =
+                std::make_shared<vectorized::ChunkExtraColumnsData>(_meta.extra_data_metas, std::move(extra_columns));
```

Comment:
Why not move these code to `ChunkExtraColumnsData` ?

Reply:
`_meta.extra_data_metas` is copied to each Chunk and to be reused by the later Chunk.